include/dlaf/sender/transform_mpi.h

```diff
@@ -90,10 +119,26 @@ template <typename F, typename Sender,
           typename = std::enable_if_t<pika::execution::experimental::is_sender_v<Sender>>>
 [[nodiscard]] decltype(auto) transformMPI(F&& f, Sender&& sender) {
   namespace ex = pika::execution::experimental;
-
-  return ex::transfer(std::forward<Sender>(sender), dlaf::internal::getMPIScheduler()) |
-         ex::then(dlaf::common::internal::ConsumeRvalues{MPICallHelper{std::forward<F>(f)}}) |
-         ex::drop_operation_state();
+  namespace mpi = pika::mpi::experimental;
+  namespace mpid = pika::mpi::experimental::detail;
+
+#ifdef EXTRA_MPI_TYPES_DEBUGGING
+  auto snd1 =
+      std::forward<Sender>(sender) |
+      ex::let_value([=, f = std::move(f)]<typename... LArgs>(LArgs&&... largs) {
+        PIKA_DETAIL_DP(dla_debug<2>, debug(str<>("Args to MPI fn\n"),
+                                           pika::debug::print_type<LArgs...>(", "), "\nValues\n"));
+        return ex::just(std::move(largs)...) |
+               mpi::transform_mpi(dlaf::common::internal::ConsumeRvalues{MPICallHelper{std::move(f)}});
+      });
+  return ex::make_unique_any_sender(std::move(snd1));
+#else
+  PIKA_DETAIL_DP(dla_debug<5>, debug(str<>("MPI fn\n")));
+  auto snd1 =
+      std::forward<Sender>(sender) |
+      mpi::transform_mpi(dlaf::common::internal::ConsumeRvalues{MPICallHelper{std::forward<F>(f)}});
+  return ex::make_unique_any_sender(std::move(snd1));
+#endif
```

Comment:
```suggestion
  return ex::make_unique_any_sender(
      std::forward<Sender>(sender) |
      mpi::transform_mpi(dlaf::common::internal::ConsumeRvalues{MPICallHelper{std::forward<F>(f)}}));
```
(modulo formatting).